Framework/Geometry/src/Crystal/CrystalStructure.cpp

```diff
@@ -82,11 +82,7 @@ void CrystalStructure::setScatterers(const CompositeBraggScatterer_sptr &scatter
 /// Adds all scatterers in the supplied collection into the internal one
 /// (scatterers are copied).
 void CrystalStructure::addScatterers(const CompositeBraggScatterer_sptr &scatterers) {
-  size_t count = scatterers->nScatterers();
-
-  for (size_t i = 0; i < count; ++i) {
-    m_scatterers->addScatterer(scatterers->getScatterer(i));
-  }
+  m_scatterers->setScatterers(scatterers->getScatterers());
```

Comment:
I notice the `addScatterer()` function does not call `removeAllScatterers()`, while `setScatterers()` does.  It might be safer to define inside `CompositeBraggScatterer` a new function `addScatterers()` which has this line, then update `setScatterers()` to use the new `addScatterers()` function.

Reply:
Suppose a user has a `CrystalStructure` with some scatterers, and wishes to add more.  The user can see in L75 above there is a `setScatterers()` method which resets the list.  That is not what the user wants, so the user goes with `addScatterers()`.  Unbeknownst to the user, this method does the exact same thing as `setScatterers()` and first resets the list.  To retain original behavior and be more clear, this file should have:
``` c++
void CrystalStructure::setScatterers(const CompositeBraggScatterer_sptr &scatterers) {
  m_scatterers->removeAllScatterers();
  addScatterers(scatterers);
}

void CrystalStructure::addScatterers(const CompositeBraggScatterer_sptr &scatterers) {
  m_scatterers->addScatterers(scatterers->getScatterers());
  assignUnitCellToScatterers(m_cell);
}
```
and then `CompositeBraggScatterer.cpp` should have
``` c++
void CompositeBraggScatterer::addScatterer(const BraggScatterer_sptr &scatterer) {
  addScattererImplementation(scatterer);
  redeclareProperties();
}

void CompositeBraggScatterer::addScatterers(const std::vector<BraggScatterer_sptr> &scatterers) {
  for (const auto &scatterer : scatterers) {
    addScattererImplementation(scatterer);
  }
  redeclareProperties();
}

void CompositeBraggScatterer::setScatterers(const std::vector<BraggScatterer_sptr> &scatterers) {
  removeAllScatterers();
  addScatterers(scatterers);
}
```